returncheck.go

```diff
@@ -0,0 +1,26 @@
+package returncheck
+
+import (
+	"flag"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:  "returncheck",
+	Doc:   "Go analyzer that requires function return values be used or explicitly ignored with `_`",
+	URL:   "https://github.com/navijation/go-returncheck",
+	Flags: flag.FlagSet{},
+	Run: func(*analysis.Pass) (interface{}, error) {
+		return nil, nil
+	},
+	// TODO: it should actually be possible to run this analyzer in spite of type errors, although
```

Comment:
If there a way to not run this for specific type errors like wrong interface implementations? My thought here is if a method is using an outdated/wrong function signature we may not run the analyzer or it should also surface this error

Reply:
I'm not sure it's possible to see what kinds of errors there were during parsing. But basically my idea here is to try to fallback to the simplest logic, which is:

1. Check to see if an expression statement wraps a call expression
2. If so, try to lookup the return signature of the function using `pass.TypesInfo.TypeOf`
3. If so, check to see if returns any values and report an error

If the second step fails, we can report nothing and move on.

Reply:
I'm not sure it's possible to see what kinds of errors there were during parsing. But I don't think incorrect interface implementations should affect things. Ultimately it's a question of looking at a single naked function call and looking at the signature of that one function. If the caller forgot to update the underlying type for an interface, it doesn't change the fact that this one call site (presumably a call to an interface) is problematic.